orderBook.js

```diff
@@ -1,19 +1,46 @@
+const removeOrderByIndex = (existingBook, orderIndex) => [
+  ...existingBook.slice(0, orderIndex),
+  ...existingBook.slice(orderIndex + 1)
+]
+
+const matchBuyAndSell = (typeOne, typeTwo) => (
+  (typeOne === 'buy' && typeTwo === 'sell') ||
+  (typeOne === 'sell' && typeTwo === 'buy')
+)
+
 const reconcileOrder = (existingBook, incomingOrder) => {
-  const foundIndex = existingBook.findIndex({
-    //need a function that looks at the parts of the incoming order and changes the type looked for to buy
-    ...incomingOrder,
-    type: 'buy'
+  const orderIndex = existingBook.findIndex(order => {
+    return (
+      order.price === incomingOrder.price &&
```

Comment:
This is where you'd have to tackle extra credit. Take a look at my functional-solution branch, I have this broken down slightly differently and it allows for solving the extra credit with a single ternary because the price match is its own function.